### ISSUE TITLE
Enter Date feature is not working

### ISSUE THREAD

**Issue by anuradhagupta345** [zulip/zulip-android#303]
When one select certain date from `Enter Date` menu then it is showing **Sorry, no messages here.** but when I scroll manually to that certain date then I find certain messages there.
I tried it many times but every time it is giving the same message. I think there is a bug in _Enter Date_ feature!

@Sam1301 @kunall17 @timabbott Can I work on this?

**Comment by kunall17** [comment]
@vishwesh3

**Comment by vishwesh3** [comment]
Thanks @anuradhagupta345 for reporting this.
The problem is not in **Enter Date** Implementation. I tried to figure out this and found that problem also persists in **Today** , in general problem is in Narrowing to date (whether it is today's date or previous date). So can we change title to Bug in Narrowing to any date.
![videotogif_2017 01 16_19 52 54](https://cloud.githubusercontent.com/assets/18511177/21986880/494148d2-dc27-11e6-9380-16afafd881df.gif)
 
As per current implementation on narrowing we check for local database for messages, and show messages which are associated with that date.

In @anuradhagupta345 case, I think app is recently installed so on home previous messages are not loaded and eventually they are not stored in database. So on narrowing it showed **Sorry, no messages here**. Once back dated messages are stored **Enter Date** feature will work fine.  
My app and @kunall17  app's  databases contains previous dates messages too, so it is working absolutely fine and were not able to point out this that time. 

Basically on narrowing we filter messages from local database and fetch 100 messages. What if entered date is of 15 days back there are more than 100 messages in this days, it will load 100 but will not able to filter that to that date. 
To overcome this issue we have to load messages of that date from server on narrowing to that date (in where clause like we do on narrowing to private messages https://github.com/zulip/zulip-android/blob/master/app/src/main/java/com/zulip/android/filters/NarrowFilterPM.java#L107).
 
https://github.com/zulip/zulip-android/blob/master/app/src/main/java/com/zulip/android/filters/NarrowFilterByDate.java#L91
and https://github.com/zulip/zulip-android/blob/master/app/src/main/java/com/zulip/android/filters/NarrowFilterByDate.java#L96
while fetching https://github.com/zulip/zulip-android/blob/master/app/src/main/java/com/zulip/android/networking/AsyncGetOldMessages.java#L208 

So what is the attribute for timestamp on server's DB (like 'pm-with' in case of NarrowFilterPM) which can be used in where clause? @kunall17 Am I missing out something or this is the case?

**Comment by kunall17** [comment]
I think the server doesn't handle this! 
@timabbott Is there a parameter for the v1/messages path to get messages from a particular date?

**Comment by timabbott** [comment]
Nope.  That's a feature that isn't that interesting on web, so we haven't done it before.  I think we actually don't have a database index on the date, even, though that's something we could change.  

Maybe open an issue on the Zulip server project for adding support for such a narrow?
